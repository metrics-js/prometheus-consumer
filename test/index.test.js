'use strict';

const { Readable } = require('readable-stream');

const promClient = require('prom-client');
const PrometheusMetricsConsumer = require('../lib');

const src = arr =>
    new Readable({
        objectMode: true,
        read() {
            arr.forEach(el => {
                this.push(el);
            });
            this.push(null);
        },
    });

test('toString returns correct object name', () => {
    expect(
        new PrometheusMetricsConsumer({ client: promClient }).toString(),
    ).toBe('[object PrometheusMetricsConsumer]');
});

test('metrics interpreted as a counter', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    const source = src([
        { name: 'my_counter', description: '.' },
        { name: 'my_counter', description: '.' },
        { name: 'my_counter', description: '.' },
    ]);

    source.pipe(consumer);

    consumer.on('finish', () => {
        expect(
            consumer.registry.getSingleMetric('my_counter'),
        ).toMatchSnapshot();
        done();
    });
});

test('metrics interpreted as a histogram', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    const source = src([
        { name: 'my_histogram', description: '.', time: 0.1 },
        { name: 'my_histogram', description: '.', time: 0.1 },
        { name: 'my_histogram', description: '.', time: 0.2 },
        { name: 'my_histogram', description: '.', time: 0.3 },
        { name: 'my_histogram', description: '.', time: 0.2 },
        { name: 'my_histogram', description: '.', time: 0.9 },
    ]);

    source.pipe(consumer);

    consumer.on('finish', () => {
        expect(
            consumer.registry.getSingleMetric('my_histogram'),
        ).toMatchSnapshot();
        done();
    });
});

test('gauge metrics interpreted as counters', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    const source = src([
        { name: 'my_gauge', description: '.', value: 10 },
        { name: 'my_gauge', description: '.', value: 8 },
    ]);

    source.pipe(consumer);

    consumer.on('finish', () => {
        expect(consumer.registry.getSingleMetric('my_gauge')).toMatchSnapshot();
        done();
    });
});

test('mixed metrics interpreted correctly', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    const source = src([
        { name: 'mixed_counter', description: '.', timestamp: 12324355 },
        {
            name: 'mixed_gauge',
            description: '.',
            value: 8,
            timestamp: 12324356,
        },
        {
            name: 'mixed_histogram',
            description: '.',
            time: 0.1,
            timestamp: 12324357,
        },
        { name: 'mixed_counter', description: '.', timestamp: 12324358 },
        {
            name: 'mixed_gauge',
            description: '.',
            value: 4,
            timestamp: 12324359,
        },
        {
            name: 'mixed_histogram',
            description: '.',
            time: 0.2,
            timestamp: 123243510,
        },
        { name: 'mixed_counter', description: '.', timestamp: 123243511 },
        {
            name: 'mixed_gauge',
            description: '.',
            value: 10,
            timestamp: 123243512,
        },
        {
            name: 'mixed_histogram',
            description: '.',
            time: 0.2,
            timestamp: 123243513,
        },
        {
            name: 'new_summary',
            description: '.',
            value: 0.2,
            timestamp: 123243513,
            type: 7,
        },
        {
            name: 'new_histogram',
            description: '.',
            value: 0.2,
            timestamp: 123243513,
            type: 5,
        },
        {
            name: 'new_gauge',
            description: '.',
            value: 0.2,
            timestamp: 123243513,
            type: 1,
        },
        {
            name: 'new_counter',
            description: '.',
            value: 0.2,
            timestamp: 123243513,
            type: 2,
        },
    ]);

    source.pipe(consumer);

    consumer.on('finish', () => {
        // eslint-disable-next-line no-underscore-dangle
        expect(consumer.registry._metrics).toMatchSnapshot();
        done();
    });
});

test('time metrics overridden to be summaries', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    consumer.override('time_series', { type: 'summary' });

    const source = src([
        {
            name: 'time_series',
            description: '.',
            time: 10,
            timestamp: 12324357,
        },
        {
            name: 'time_series',
            description: '.',
            time: 12,
            timestamp: 123243510,
        },
        {
            name: 'time_series',
            description: '.',
            time: 1,
            timestamp: 123243513,
        },
        {
            name: 'time_series',
            description: '.',
            time: 10,
            timestamp: 123243514,
        },
        {
            name: 'time_series',
            description: '.',
            time: 11,
            timestamp: 123243515,
        },
        {
            name: 'time_series',
            description: '.',
            time: 12,
            timestamp: 123243516,
        },
    ]);

    source.pipe(consumer);

    consumer.on('finish', () => {
        // eslint-disable-next-line no-underscore-dangle
        expect(consumer.registry._metrics.time_series).toBeInstanceOf(
            promClient.Summary,
        );
        done();
    });
});

test('metrics with labels', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    const source = src([
        {
            name: 'my_counter_with_labels',
            description: '.',
            podlet: 'recommendations',
            url: 'http://mylayout.com',
            method: 'GET',
            status: 200,
            meta: {
                label1: 'one',
                label2: 'two',
            },
        },
    ]);

    source.pipe(consumer);

    consumer.on('finish', () => {
        expect(
            consumer.registry.getSingleMetric('my_counter_with_labels'),
        ).toMatchSnapshot();
        done();
    });
});

test('invalid constructor options', () => {
    expect.hasAssertions();
    expect(
        () =>
            new PrometheusMetricsConsumer({ client: promClient, fake: 'key' }),
    ).toThrowError();
});

test('.override() missing name', () => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });
    expect(() => {
        consumer.override();
    }).toThrowErrorMatchingSnapshot();
});

test('.override() invalid name', () => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });
    expect(() => {
        consumer.override('my name is bad');
    }).toThrowErrorMatchingSnapshot();
});

test('.override() invalid type', () => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });
    expect(() => {
        consumer.override('valid_name', { type: 'not allowed type' });
    }).toThrowErrorMatchingSnapshot();
});

test('.override() sets a metric to be a counter', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });
    const source = src([{ name: 'valid_name', description: '.', time: 12345 }]);

    consumer.override('valid_name', { type: 'counter' });
    source.pipe(consumer);

    consumer.on('finish', () => {
        expect(
            consumer.registry.getSingleMetric('valid_name'),
        ).toMatchSnapshot();
        done();
    });
});

test('.override() sets a metric to be a histogram with specific buckets', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });
    const source = src([
        {
            name: 'overridden_histogram',
            description: '.',
            time: 12345,
        },
    ]);

    consumer.override('overridden_histogram', {
        type: 'histogram',
        buckets: {
            bucketStepStart: 1,
            bucketStepFactor: 2,
            bucketStepCount: 8,
        },
    });
    source.pipe(consumer);

    consumer.on('finish', () => {
        expect(
            consumer.registry.getSingleMetric('overridden_histogram'),
        ).toMatchSnapshot();
        done();
    });
});

test('.override() sets a counter with custom labels', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    const source = src([
        {
            name: 'custom_label_counter',
            description: '.',
            url: 'http://example.com',
            method: 'GET',
        },
    ]);

    consumer.on('finish', () => {
        expect(
            consumer.registry.getSingleMetric('custom_label_counter'),
        ).toMatchSnapshot();
        done();
    });

    consumer.override('custom_label_counter', {
        type: 'counter',
    });
    source.pipe(consumer);
});

test('.override() sets a histogram with custom labels', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });

    const source = src([
        {
            name: 'custom_label_histogram',
            description: '.',
            time: 12324,
            url: 'http://example.com',
            method: 'GET',
        },
    ]);

    consumer.on('finish', () => {
        expect(
            consumer.registry.getSingleMetric('custom_label_histogram'),
        ).toMatchSnapshot();
        done();
    });

    consumer.override('custom_label_histogram', {
        type: 'histogram',
    });
    source.pipe(consumer);
});

test('.contentType() method', () => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });
    expect(consumer.contentType()).toBe(
        'text/plain; version=0.0.4; charset=utf-8',
    );
});

test('.metrics() method', done => {
    expect.hasAssertions();
    const consumer = new PrometheusMetricsConsumer({ client: promClient });
    const source = src([
        {
            name: 'custom_label_histogram',
            description: '.',
            time: 12324,
            url: 'http://example.com',
            method: 'GET',
        },
    ]);

    consumer.on('finish', () => {
        expect(consumer.metrics()).toMatchSnapshot();
        done();
    });
    source.pipe(consumer);
});
