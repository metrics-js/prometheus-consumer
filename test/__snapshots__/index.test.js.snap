// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.metrics() method 1`] = `
"# HELP custom_label_histogram .
# TYPE custom_label_histogram histogram
custom_label_histogram_bucket{le=\\"0.001\\"} 0
custom_label_histogram_bucket{le=\\"0.0012\\"} 0
custom_label_histogram_bucket{le=\\"0.0013\\"} 0
custom_label_histogram_bucket{le=\\"0.0015\\"} 0
custom_label_histogram_bucket{le=\\"0.0017\\"} 0
custom_label_histogram_bucket{le=\\"0.002\\"} 0
custom_label_histogram_bucket{le=\\"0.0023\\"} 0
custom_label_histogram_bucket{le=\\"0.0027\\"} 0
custom_label_histogram_bucket{le=\\"0.0031\\"} 0
custom_label_histogram_bucket{le=\\"0.0035\\"} 0
custom_label_histogram_bucket{le=\\"0.004\\"} 0
custom_label_histogram_bucket{le=\\"0.0047\\"} 0
custom_label_histogram_bucket{le=\\"0.0054\\"} 0
custom_label_histogram_bucket{le=\\"0.0062\\"} 0
custom_label_histogram_bucket{le=\\"0.0071\\"} 0
custom_label_histogram_bucket{le=\\"0.0081\\"} 0
custom_label_histogram_bucket{le=\\"0.0094\\"} 0
custom_label_histogram_bucket{le=\\"0.0108\\"} 0
custom_label_histogram_bucket{le=\\"0.0124\\"} 0
custom_label_histogram_bucket{le=\\"0.0142\\"} 0
custom_label_histogram_bucket{le=\\"0.0164\\"} 0
custom_label_histogram_bucket{le=\\"0.0188\\"} 0
custom_label_histogram_bucket{le=\\"0.0216\\"} 0
custom_label_histogram_bucket{le=\\"0.0249\\"} 0
custom_label_histogram_bucket{le=\\"0.0286\\"} 0
custom_label_histogram_bucket{le=\\"0.0329\\"} 0
custom_label_histogram_bucket{le=\\"0.0379\\"} 0
custom_label_histogram_bucket{le=\\"0.0435\\"} 0
custom_label_histogram_bucket{le=\\"0.0501\\"} 0
custom_label_histogram_bucket{le=\\"0.0576\\"} 0
custom_label_histogram_bucket{le=\\"0.0662\\"} 0
custom_label_histogram_bucket{le=\\"0.0761\\"} 0
custom_label_histogram_bucket{le=\\"0.0876\\"} 0
custom_label_histogram_bucket{le=\\"0.1007\\"} 0
custom_label_histogram_bucket{le=\\"0.1158\\"} 0
custom_label_histogram_bucket{le=\\"0.1332\\"} 0
custom_label_histogram_bucket{le=\\"0.1532\\"} 0
custom_label_histogram_bucket{le=\\"0.1761\\"} 0
custom_label_histogram_bucket{le=\\"0.2025\\"} 0
custom_label_histogram_bucket{le=\\"0.2329\\"} 0
custom_label_histogram_bucket{le=\\"0.2679\\"} 0
custom_label_histogram_bucket{le=\\"0.308\\"} 0
custom_label_histogram_bucket{le=\\"0.3542\\"} 0
custom_label_histogram_bucket{le=\\"0.4074\\"} 0
custom_label_histogram_bucket{le=\\"0.4685\\"} 0
custom_label_histogram_bucket{le=\\"Infinity\\"} 1
custom_label_histogram_bucket{le=\\"+Inf\\"} 1
custom_label_histogram_sum 12324
custom_label_histogram_count 1
"
`;

exports[`.override() invalid name 1`] = `"Invalid \`name\` given to prometheusMetricsConsumer.override. \\"name\\" must only contain alpha-numeric and underscore characters"`;

exports[`.override() invalid type 1`] = `
"Invalid \`config\` object given to prometheusMetricsConsumer.override. {
  \\"type\\" [31m[1][0m: \\"not allowed type\\"
}
[31m
[1] \\"type\\" must be one of [counter, histogram, gauge, summary][0m"
`;

exports[`.override() missing name 1`] = `"Invalid \`name\` given to prometheusMetricsConsumer.override. \\"name\\" is required"`;

exports[`.override() sets a counter with custom labels 1`] = `
Counter {
  "aggregator": "sum",
  "hashMap": Object {
    "": Object {
      "labels": Object {},
      "timestamp": undefined,
      "value": 1,
    },
  },
  "help": ".",
  "labelNames": Array [],
  "name": "custom_label_counter",
}
`;

exports[`.override() sets a histogram with custom labels 1`] = `
Histogram {
  "aggregator": "sum",
  "bucketValues": Object {
    "0.001": 0,
    "0.0012": 0,
    "0.0013": 0,
    "0.0015": 0,
    "0.0017": 0,
    "0.002": 0,
    "0.0023": 0,
    "0.0027": 0,
    "0.0031": 0,
    "0.0035": 0,
    "0.004": 0,
    "0.0047": 0,
    "0.0054": 0,
    "0.0062": 0,
    "0.0071": 0,
    "0.0081": 0,
    "0.0094": 0,
    "0.0108": 0,
    "0.0124": 0,
    "0.0142": 0,
    "0.0164": 0,
    "0.0188": 0,
    "0.0216": 0,
    "0.0249": 0,
    "0.0286": 0,
    "0.0329": 0,
    "0.0379": 0,
    "0.0435": 0,
    "0.0501": 0,
    "0.0576": 0,
    "0.0662": 0,
    "0.0761": 0,
    "0.0876": 0,
    "0.1007": 0,
    "0.1158": 0,
    "0.1332": 0,
    "0.1532": 0,
    "0.1761": 0,
    "0.2025": 0,
    "0.2329": 0,
    "0.2679": 0,
    "0.308": 0,
    "0.3542": 0,
    "0.4074": 0,
    "0.4685": 0,
    "Infinity": 0,
  },
  "count": 0,
  "hashMap": Object {
    "": Object {
      "bucketValues": Object {
        "0.001": 0,
        "0.0012": 0,
        "0.0013": 0,
        "0.0015": 0,
        "0.0017": 0,
        "0.002": 0,
        "0.0023": 0,
        "0.0027": 0,
        "0.0031": 0,
        "0.0035": 0,
        "0.004": 0,
        "0.0047": 0,
        "0.0054": 0,
        "0.0062": 0,
        "0.0071": 0,
        "0.0081": 0,
        "0.0094": 0,
        "0.0108": 0,
        "0.0124": 0,
        "0.0142": 0,
        "0.0164": 0,
        "0.0188": 0,
        "0.0216": 0,
        "0.0249": 0,
        "0.0286": 0,
        "0.0329": 0,
        "0.0379": 0,
        "0.0435": 0,
        "0.0501": 0,
        "0.0576": 0,
        "0.0662": 0,
        "0.0761": 0,
        "0.0876": 0,
        "0.1007": 0,
        "0.1158": 0,
        "0.1332": 0,
        "0.1532": 0,
        "0.1761": 0,
        "0.2025": 0,
        "0.2329": 0,
        "0.2679": 0,
        "0.308": 0,
        "0.3542": 0,
        "0.4074": 0,
        "0.4685": 0,
        "Infinity": 1,
      },
      "count": 1,
      "labels": Object {},
      "sum": 12324,
    },
  },
  "help": ".",
  "labelNames": Array [],
  "name": "custom_label_histogram",
  "sum": 0,
  "upperBounds": Array [
    0.001,
    0.0012,
    0.0013,
    0.0015,
    0.0017,
    0.002,
    0.0023,
    0.0027,
    0.0031,
    0.0035,
    0.004,
    0.0047,
    0.0054,
    0.0062,
    0.0071,
    0.0081,
    0.0094,
    0.0108,
    0.0124,
    0.0142,
    0.0164,
    0.0188,
    0.0216,
    0.0249,
    0.0286,
    0.0329,
    0.0379,
    0.0435,
    0.0501,
    0.0576,
    0.0662,
    0.0761,
    0.0876,
    0.1007,
    0.1158,
    0.1332,
    0.1532,
    0.1761,
    0.2025,
    0.2329,
    0.2679,
    0.308,
    0.3542,
    0.4074,
    0.4685,
    Infinity,
  ],
}
`;

exports[`.override() sets a metric to be a counter 1`] = `
Counter {
  "aggregator": "sum",
  "hashMap": Object {
    "": Object {
      "labels": Object {},
      "timestamp": undefined,
      "value": 1,
    },
  },
  "help": ".",
  "labelNames": Array [],
  "name": "valid_name",
}
`;

exports[`.override() sets a metric to be a histogram with specific buckets 1`] = `
Histogram {
  "aggregator": "sum",
  "bucketValues": Object {
    "1": 0,
    "128": 0,
    "16": 0,
    "2": 0,
    "32": 0,
    "4": 0,
    "64": 0,
    "8": 0,
    "Infinity": 0,
  },
  "count": 0,
  "hashMap": Object {
    "": Object {
      "bucketValues": Object {
        "1": 0,
        "128": 0,
        "16": 0,
        "2": 0,
        "32": 0,
        "4": 0,
        "64": 0,
        "8": 0,
        "Infinity": 1,
      },
      "count": 1,
      "labels": Object {},
      "sum": 12345,
    },
  },
  "help": ".",
  "labelNames": Array [],
  "name": "overridden_histogram",
  "sum": 0,
  "upperBounds": Array [
    1,
    2,
    4,
    8,
    16,
    32,
    64,
    128,
    Infinity,
  ],
}
`;

exports[`gauge metrics interpreted as counters 1`] = `
Gauge {
  "aggregator": "sum",
  "hashMap": Object {
    "": Object {
      "labels": Object {},
      "timestamp": undefined,
      "value": 8,
    },
  },
  "help": ".",
  "labelNames": Array [],
  "name": "my_gauge",
}
`;

exports[`metrics interpreted as a counter 1`] = `
Counter {
  "aggregator": "sum",
  "hashMap": Object {
    "": Object {
      "labels": Object {},
      "timestamp": undefined,
      "value": 3,
    },
  },
  "help": ".",
  "labelNames": Array [],
  "name": "my_counter",
}
`;

exports[`metrics interpreted as a histogram 1`] = `
Histogram {
  "aggregator": "sum",
  "bucketValues": Object {
    "0.001": 0,
    "0.0012": 0,
    "0.0013": 0,
    "0.0015": 0,
    "0.0017": 0,
    "0.002": 0,
    "0.0023": 0,
    "0.0027": 0,
    "0.0031": 0,
    "0.0035": 0,
    "0.004": 0,
    "0.0047": 0,
    "0.0054": 0,
    "0.0062": 0,
    "0.0071": 0,
    "0.0081": 0,
    "0.0094": 0,
    "0.0108": 0,
    "0.0124": 0,
    "0.0142": 0,
    "0.0164": 0,
    "0.0188": 0,
    "0.0216": 0,
    "0.0249": 0,
    "0.0286": 0,
    "0.0329": 0,
    "0.0379": 0,
    "0.0435": 0,
    "0.0501": 0,
    "0.0576": 0,
    "0.0662": 0,
    "0.0761": 0,
    "0.0876": 0,
    "0.1007": 0,
    "0.1158": 0,
    "0.1332": 0,
    "0.1532": 0,
    "0.1761": 0,
    "0.2025": 0,
    "0.2329": 0,
    "0.2679": 0,
    "0.308": 0,
    "0.3542": 0,
    "0.4074": 0,
    "0.4685": 0,
    "Infinity": 0,
  },
  "count": 0,
  "hashMap": Object {
    "": Object {
      "bucketValues": Object {
        "0.001": 0,
        "0.0012": 0,
        "0.0013": 0,
        "0.0015": 0,
        "0.0017": 0,
        "0.002": 0,
        "0.0023": 0,
        "0.0027": 0,
        "0.0031": 0,
        "0.0035": 0,
        "0.004": 0,
        "0.0047": 0,
        "0.0054": 0,
        "0.0062": 0,
        "0.0071": 0,
        "0.0081": 0,
        "0.0094": 0,
        "0.0108": 0,
        "0.0124": 0,
        "0.0142": 0,
        "0.0164": 0,
        "0.0188": 0,
        "0.0216": 0,
        "0.0249": 0,
        "0.0286": 0,
        "0.0329": 0,
        "0.0379": 0,
        "0.0435": 0,
        "0.0501": 0,
        "0.0576": 0,
        "0.0662": 0,
        "0.0761": 0,
        "0.0876": 0,
        "0.1007": 2,
        "0.1158": 0,
        "0.1332": 0,
        "0.1532": 0,
        "0.1761": 0,
        "0.2025": 2,
        "0.2329": 0,
        "0.2679": 0,
        "0.308": 1,
        "0.3542": 0,
        "0.4074": 0,
        "0.4685": 0,
        "Infinity": 1,
      },
      "count": 6,
      "labels": Object {},
      "sum": 1.7999999999999998,
    },
  },
  "help": ".",
  "labelNames": Array [],
  "name": "my_histogram",
  "sum": 0,
  "upperBounds": Array [
    0.001,
    0.0012,
    0.0013,
    0.0015,
    0.0017,
    0.002,
    0.0023,
    0.0027,
    0.0031,
    0.0035,
    0.004,
    0.0047,
    0.0054,
    0.0062,
    0.0071,
    0.0081,
    0.0094,
    0.0108,
    0.0124,
    0.0142,
    0.0164,
    0.0188,
    0.0216,
    0.0249,
    0.0286,
    0.0329,
    0.0379,
    0.0435,
    0.0501,
    0.0576,
    0.0662,
    0.0761,
    0.0876,
    0.1007,
    0.1158,
    0.1332,
    0.1532,
    0.1761,
    0.2025,
    0.2329,
    0.2679,
    0.308,
    0.3542,
    0.4074,
    0.4685,
    Infinity,
  ],
}
`;

exports[`metrics with labels 1`] = `
Counter {
  "aggregator": "sum",
  "hashMap": Object {
    "label1:one,label2:two": Object {
      "labels": Object {
        "label1": "one",
        "label2": "two",
      },
      "timestamp": undefined,
      "value": 1,
    },
  },
  "help": ".",
  "labelNames": Array [
    "label1",
    "label2",
  ],
  "name": "my_counter_with_labels",
}
`;

exports[`mixed metrics interpreted correctly 1`] = `
Object {
  "mixed_counter": Counter {
    "aggregator": "sum",
    "hashMap": Object {
      "": Object {
        "labels": Object {},
        "timestamp": undefined,
        "value": 3,
      },
    },
    "help": ".",
    "labelNames": Array [],
    "name": "mixed_counter",
  },
  "mixed_gauge": Gauge {
    "aggregator": "sum",
    "hashMap": Object {
      "": Object {
        "labels": Object {},
        "timestamp": undefined,
        "value": 10,
      },
    },
    "help": ".",
    "labelNames": Array [],
    "name": "mixed_gauge",
  },
  "mixed_histogram": Histogram {
    "aggregator": "sum",
    "bucketValues": Object {
      "0.001": 0,
      "0.0012": 0,
      "0.0013": 0,
      "0.0015": 0,
      "0.0017": 0,
      "0.002": 0,
      "0.0023": 0,
      "0.0027": 0,
      "0.0031": 0,
      "0.0035": 0,
      "0.004": 0,
      "0.0047": 0,
      "0.0054": 0,
      "0.0062": 0,
      "0.0071": 0,
      "0.0081": 0,
      "0.0094": 0,
      "0.0108": 0,
      "0.0124": 0,
      "0.0142": 0,
      "0.0164": 0,
      "0.0188": 0,
      "0.0216": 0,
      "0.0249": 0,
      "0.0286": 0,
      "0.0329": 0,
      "0.0379": 0,
      "0.0435": 0,
      "0.0501": 0,
      "0.0576": 0,
      "0.0662": 0,
      "0.0761": 0,
      "0.0876": 0,
      "0.1007": 0,
      "0.1158": 0,
      "0.1332": 0,
      "0.1532": 0,
      "0.1761": 0,
      "0.2025": 0,
      "0.2329": 0,
      "0.2679": 0,
      "0.308": 0,
      "0.3542": 0,
      "0.4074": 0,
      "0.4685": 0,
      "Infinity": 0,
    },
    "count": 0,
    "hashMap": Object {
      "": Object {
        "bucketValues": Object {
          "0.001": 0,
          "0.0012": 0,
          "0.0013": 0,
          "0.0015": 0,
          "0.0017": 0,
          "0.002": 0,
          "0.0023": 0,
          "0.0027": 0,
          "0.0031": 0,
          "0.0035": 0,
          "0.004": 0,
          "0.0047": 0,
          "0.0054": 0,
          "0.0062": 0,
          "0.0071": 0,
          "0.0081": 0,
          "0.0094": 0,
          "0.0108": 0,
          "0.0124": 0,
          "0.0142": 0,
          "0.0164": 0,
          "0.0188": 0,
          "0.0216": 0,
          "0.0249": 0,
          "0.0286": 0,
          "0.0329": 0,
          "0.0379": 0,
          "0.0435": 0,
          "0.0501": 0,
          "0.0576": 0,
          "0.0662": 0,
          "0.0761": 0,
          "0.0876": 0,
          "0.1007": 1,
          "0.1158": 0,
          "0.1332": 0,
          "0.1532": 0,
          "0.1761": 0,
          "0.2025": 2,
          "0.2329": 0,
          "0.2679": 0,
          "0.308": 0,
          "0.3542": 0,
          "0.4074": 0,
          "0.4685": 0,
          "Infinity": 0,
        },
        "count": 3,
        "labels": Object {},
        "sum": 0.5,
      },
    },
    "help": ".",
    "labelNames": Array [],
    "name": "mixed_histogram",
    "sum": 0,
    "upperBounds": Array [
      0.001,
      0.0012,
      0.0013,
      0.0015,
      0.0017,
      0.002,
      0.0023,
      0.0027,
      0.0031,
      0.0035,
      0.004,
      0.0047,
      0.0054,
      0.0062,
      0.0071,
      0.0081,
      0.0094,
      0.0108,
      0.0124,
      0.0142,
      0.0164,
      0.0188,
      0.0216,
      0.0249,
      0.0286,
      0.0329,
      0.0379,
      0.0435,
      0.0501,
      0.0576,
      0.0662,
      0.0761,
      0.0876,
      0.1007,
      0.1158,
      0.1332,
      0.1532,
      0.1761,
      0.2025,
      0.2329,
      0.2679,
      0.308,
      0.3542,
      0.4074,
      0.4685,
      Infinity,
    ],
  },
  "new_counter": Counter {
    "aggregator": "sum",
    "hashMap": Object {
      "": Object {
        "labels": Object {},
        "timestamp": undefined,
        "value": 0.2,
      },
    },
    "help": ".",
    "labelNames": Array [],
    "name": "new_counter",
  },
  "new_gauge": Gauge {
    "aggregator": "sum",
    "hashMap": Object {
      "": Object {
        "labels": Object {},
        "timestamp": undefined,
        "value": 0.2,
      },
    },
    "help": ".",
    "labelNames": Array [],
    "name": "new_gauge",
  },
  "new_histogram": Histogram {
    "aggregator": "sum",
    "bucketValues": Object {
      "0.001": 0,
      "0.0012": 0,
      "0.0013": 0,
      "0.0015": 0,
      "0.0017": 0,
      "0.002": 0,
      "0.0023": 0,
      "0.0027": 0,
      "0.0031": 0,
      "0.0035": 0,
      "0.004": 0,
      "0.0047": 0,
      "0.0054": 0,
      "0.0062": 0,
      "0.0071": 0,
      "0.0081": 0,
      "0.0094": 0,
      "0.0108": 0,
      "0.0124": 0,
      "0.0142": 0,
      "0.0164": 0,
      "0.0188": 0,
      "0.0216": 0,
      "0.0249": 0,
      "0.0286": 0,
      "0.0329": 0,
      "0.0379": 0,
      "0.0435": 0,
      "0.0501": 0,
      "0.0576": 0,
      "0.0662": 0,
      "0.0761": 0,
      "0.0876": 0,
      "0.1007": 0,
      "0.1158": 0,
      "0.1332": 0,
      "0.1532": 0,
      "0.1761": 0,
      "0.2025": 0,
      "0.2329": 0,
      "0.2679": 0,
      "0.308": 0,
      "0.3542": 0,
      "0.4074": 0,
      "0.4685": 0,
      "Infinity": 0,
    },
    "count": 0,
    "hashMap": Object {
      "": Object {
        "bucketValues": Object {
          "0.001": 0,
          "0.0012": 0,
          "0.0013": 0,
          "0.0015": 0,
          "0.0017": 0,
          "0.002": 0,
          "0.0023": 0,
          "0.0027": 0,
          "0.0031": 0,
          "0.0035": 0,
          "0.004": 0,
          "0.0047": 0,
          "0.0054": 0,
          "0.0062": 0,
          "0.0071": 0,
          "0.0081": 0,
          "0.0094": 0,
          "0.0108": 0,
          "0.0124": 0,
          "0.0142": 0,
          "0.0164": 0,
          "0.0188": 0,
          "0.0216": 0,
          "0.0249": 0,
          "0.0286": 0,
          "0.0329": 0,
          "0.0379": 0,
          "0.0435": 0,
          "0.0501": 0,
          "0.0576": 0,
          "0.0662": 0,
          "0.0761": 0,
          "0.0876": 0,
          "0.1007": 0,
          "0.1158": 0,
          "0.1332": 0,
          "0.1532": 0,
          "0.1761": 0,
          "0.2025": 1,
          "0.2329": 0,
          "0.2679": 0,
          "0.308": 0,
          "0.3542": 0,
          "0.4074": 0,
          "0.4685": 0,
          "Infinity": 0,
        },
        "count": 1,
        "labels": Object {},
        "sum": 0.2,
      },
    },
    "help": ".",
    "labelNames": Array [],
    "name": "new_histogram",
    "sum": 0,
    "upperBounds": Array [
      0.001,
      0.0012,
      0.0013,
      0.0015,
      0.0017,
      0.002,
      0.0023,
      0.0027,
      0.0031,
      0.0035,
      0.004,
      0.0047,
      0.0054,
      0.0062,
      0.0071,
      0.0081,
      0.0094,
      0.0108,
      0.0124,
      0.0142,
      0.0164,
      0.0188,
      0.0216,
      0.0249,
      0.0286,
      0.0329,
      0.0379,
      0.0435,
      0.0501,
      0.0576,
      0.0662,
      0.0761,
      0.0876,
      0.1007,
      0.1158,
      0.1332,
      0.1532,
      0.1761,
      0.2025,
      0.2329,
      0.2679,
      0.308,
      0.3542,
      0.4074,
      0.4685,
      Infinity,
    ],
  },
  "new_summary": Summary {
    "ageBuckets": undefined,
    "aggregator": "sum",
    "help": ".",
    "labelNames": Array [],
    "maxAgeSeconds": undefined,
    "name": "new_summary",
    "percentiles": Array [
      0.01,
      0.05,
      0.5,
      0.9,
      0.95,
      0.99,
      0.999,
    ],
  },
}
`;
